lib/ts/recipe/dashboard/api/user/create/passwordlessUser.ts

```diff
@@ -57,39 +65,71 @@ export const createPasswordlessUser = async (
 
     if (
         email !== undefined &&
-        (passwordlessRecipe.config.contactMethod === "EMAIL" ||
-            passwordlessRecipe.config.contactMethod === "EMAIL_OR_PHONE")
+        (passwordlessOrThirdpartyPasswordlessRecipe.config.contactMethod === "EMAIL" ||
+            passwordlessOrThirdpartyPasswordlessRecipe.config.contactMethod === "EMAIL_OR_PHONE")
     ) {
         email = email.trim();
-        const validateError = await defaultValidateEmail(email);
-        if (validateError !== undefined) {
+        let validationError: string | undefined = undefined;
+
+        if (passwordlessOrThirdpartyPasswordlessRecipe.config.validateEmailAddress !== undefined) {
+            validationError = await passwordlessOrThirdpartyPasswordlessRecipe.config.validateEmailAddress(
+                email,
+                tenantId
+            );
+        } else {
+            validationError = await defaultValidateEmail(email);
+        }
+        if (validationError !== undefined) {
             return {
-                status: "INPUT_VALIDATION_ERROR",
-                message: validateError,
+                status: "EMAIL_VALIDATION_ERROR",
+                message: validationError,
             };
         }
     }
 
     if (
         phoneNumber !== undefined &&
-        (passwordlessRecipe.config.contactMethod === "PHONE" ||
-            passwordlessRecipe.config.contactMethod === "EMAIL_OR_PHONE")
+        (passwordlessOrThirdpartyPasswordlessRecipe.config.contactMethod === "PHONE" ||
+            passwordlessOrThirdpartyPasswordlessRecipe.config.contactMethod === "EMAIL_OR_PHONE")
     ) {
-        const validateError = await defaultValidatePhoneNumber(phoneNumber);
-        if (validateError !== undefined) {
+        let validationError: string | undefined = undefined;
+
+        if (passwordlessOrThirdpartyPasswordlessRecipe.config.validatePhoneNumber !== undefined) {
+            validationError = await passwordlessOrThirdpartyPasswordlessRecipe.config.validatePhoneNumber(
+                phoneNumber,
+                tenantId
+            );
+        } else {
+            validationError = await defaultValidatePhoneNumber(phoneNumber);
```

Comment:
this is wrong.. no need for this cause if validatePhoneNumber is undefined, then the contact method is email

Reply:
done